src/main/java/org/ethereum/beacon/discovery/storage/LocalNodeRecordStore.java

```diff
@@ -28,8 +30,8 @@ public NodeRecord getLocalNodeRecord() {
   public void onSocketAddressChanged(final InetSocketAddress newAddress) {
     NodeRecord oldRecord = this.latestRecord;
     NodeRecord newRecord = oldRecord.withNewAddress(newAddress, privateKey);
-    this.latestRecord = newRecord;
-    recordListener.recordUpdated(oldRecord, newRecord);
+    newAddressListener.newAddress(oldRecord, newRecord)
+            .ifPresent(record -> this.latestRecord = record);
```

Comment:
We still need to notify the `recordListener` if the record is updated.  The two functions should stay independent - one decides if/what to change, the other is notified of any changes.  Otherwise the Teku side will have to duplicate code to ensure the disk storage is updated for both events instead of being able to do it in one place.

Reply:
Good catch.